packages/backend/src/nest/ipfs/ipfs.service.ts

```diff
@@ -1,60 +1,168 @@
+import { bitswap } from '@helia/block-brokers'
 import { IPFS_REPO_PATCH } from '../const'
 import { createLogger } from '../common/logger'
+import { LevelDatastore } from 'datastore-level'
+import { LevelBlockstore, LevelBlockstoreInit } from 'blockstore-level'
+import { Libp2pService } from '../libp2p/libp2p.service'
+import { DatabaseOptions, Level } from 'level'
+
+type StoreInit = {
+  blockstore?: Omit<LevelBlockstoreInit, 'valueEncoding' | 'keyEncoding'>
+  datastore?: Omit<DatabaseOptions<string, Uint8Array>, 'valueEncoding' | 'keyEncoding'>
+}
 
 @Injectable()
 export class IpfsService {
-  public ipfsInstance: IPFS | null
-  private counter = 0
+  public ipfsInstance: Helia | null
+  private blockstore: LevelBlockstore | null
+  private datastore: LevelDatastore | null
+
+  private started: boolean
   private readonly logger = createLogger(IpfsService.name)
+
   constructor(
     @Inject(IPFS_REPO_PATCH) public readonly ipfsRepoPath: string,
-    private readonly lazyModuleLoader: LazyModuleLoader
-  ) {}
-
-  public async createInstance(peerId: any) {
-    this.counter++
-    this.logger.info('counter ipfs', this.counter)
-    const { Libp2pModule } = await import('../libp2p/libp2p.module')
-    const moduleRef = await this.lazyModuleLoader.load(() => Libp2pModule)
-    const { Libp2pService } = await import('../libp2p/libp2p.service')
-    const libp2pService = moduleRef.get(Libp2pService)
-    const libp2pInstance = libp2pService?.libp2pInstance
-
-    let ipfs: IPFS
+    private readonly libp2pService: Libp2pService
+  ) {
+    this.started = false
+  }
+
+  public async createInstance(): Promise<Helia> {
+    const libp2pInstance = this.libp2pService?.libp2pInstance
+
+    let ipfs: Helia
     try {
       if (!libp2pInstance) {
-        this.logger.error('no libp2p instance')
-        throw new Error('no libp2p instance')
+        this.logger.error('Libp2p instance required')
+        throw new Error('Libp2p instance required')
       }
-      ipfs = await create({
+
+      this.logger.info(`Initializing Helia datastore and blockstore`)
+      await this.initializeStores()
+
+      this.logger.info(`Creating Helia instance`)
+      ipfs = await createHelia({
         start: false,
-        libp2p: async () => libp2pInstance,
-        preload: { enabled: false },
-        repo: this.ipfsRepoPath,
-        EXPERIMENTAL: {
-          ipnsPubsub: true,
-        },
-        init: {
-          privateKey: peerId,
-        },
+        libp2p: libp2pInstance,
+        blockstore: this.blockstore!,
+        datastore: this.datastore!,
+        blockBrokers: [bitswap()],
       })
       this.ipfsInstance = ipfs
     } catch (error) {
-      this.logger.error('ipfs creation failed', error)
+      this.logger.error('IPFS creation failed', error)
+      throw new Error('IPFS creation failed')
     }
 
     return this.ipfsInstance
   }
 
+  private async initializeStores(init?: StoreInit): Promise<void> {
```

Comment:
This is more complicated than it should be because we need to ensure that valueEncoding is set to buffer no matter what.  I hit so many issues due to level not properly encoding to buffer...